src/main/java/org/folio/service/orders/lines/update/OrderLinePatchOperationService.java

```diff
@@ -112,6 +99,32 @@ private Future<Void> patchOrderLine(PatchOrderLineRequest request, String lineId
     return promise.future();
   }
 
+  public Future<Void> handleUpdateInstance(OrderLineUpdateInstanceHolder holder, RequestContext requestContext) {
+    PoLine storagePoLine = holder.getStoragePoLine();
+
+    var physical = PoLineCommonUtil.getPhysical(storagePoLine);
+    Future<Void> updatePhysical = physical == null ? Future.succeededFuture() :
+      orderLineUpdateInstanceStrategyResolver.resolve(CreateInventoryType.fromValue(physical.getCreateInventory().value()))
+        .updateInstance(holder, requestContext);
+
+    var eresource = PoLineCommonUtil.getEresource(storagePoLine);
+    Future<Void> updateEresource = eresource == null ? Future.succeededFuture() :
+      orderLineUpdateInstanceStrategyResolver.resolve(CreateInventoryType.fromValue(eresource.getCreateInventory().value()))
+        .updateInstance(holder, requestContext);
+
+    return updatePhysical.compose(v -> updateEresource);
```

Comment:
think about this place one more time. Currently it is hard to understand. As suggestion you can divide into two method for specific cases and use if becuease it line is too long

Reply:
Please share your opinion if it was easier to read old implementation.

Reply:
my suggestion is that using If and divide method.

Reply:
done